### PR TITLE
Add EGSEA tool for gene set testing

### PR DESCRIPTION
**Pull request by mblue9**
This PR has the first version of a wrapper for EGSEA, a tool for gene set testing (paper here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5408797/). 

This version is a wrapper around the egsea.cnt function, which can take a count matrix or count files as input (similar to the edgeR wrapper).

I'd like to add more of EGSEA in the future when I can get a chance e.g ability to input normalised counts, custom signatures, egsea.ma for microarrays...

### CODE REVIEW

File: tools/egsea/egsea.xml [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+            <output name="outReport" value="out_il13.html" compare="sim_size"/>
+            <output name="outReport">
+                <assert_contents>
+                    <has_text text="Gene Set Testing Report"/>
+                </assert_contents>
+            </output>
+            <output_collection name="outTables" count="1">
+                <element name="ranked-h-gene-sets-IL13Ant-IL13" ftype="tabular" file="ranked-h-gene-sets-IL13Ant-IL13.txt"/>
+            </output_collection>
+            <output name="outRscript" value="out_il13.txt" compare="sim_size"/>
```
Comment (bgruening) [inline tools/egsea/egsea.xml]: why should this be sim_size?

File: tools/egsea/egsea.xml [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+            <param name="format" value="matrix"/>
+            <param name="counts" value="il13.counts"/>
+            <param name="genes" value="il13.genes"/>
+            <param name="ffile" value="yes"/>
+            <param name="finfo" value="il13.group"/>
+            <repeat name="rep_contrast">
+                <param name="contrast" value="IL13Ant-IL13" />
+            </repeat>
+            <param name="rscriptOpt" value="True"/>
+            <output name="outReport" value="out_il13.html" compare="sim_size"/>
```
Comment (bgruening) [inline tools/egsea/egsea.xml]: sim_size for a text file is quite unusal.

Comment (mblue9) [reply tools/egsea/egsea.xml]: left that in from when I was testing, have removed

File: tools/egsea/egsea.xml (hunk 1 above)
Comment (mblue9) [reply tools/egsea/egsea.xml]: no reason, have removed

**Comment by mblue9** [comment]
Thanks for the review @bgruening! I've removed those sim_sizes (yes they are unnecessary) and filed the GALAXY_SLOTS bug in Galaxy here: https://github.com/galaxyproject/galaxy/issues/5369

**Comment by bgruening** [comment]
Hacks for the win!

**Pull request merged**